Makefile

```diff
@@ -1,7 +1,7 @@
 all: html
 
 
-html: py r
+html: py r j
```

Comment:
nit: why not write out `java` instead of `j`? (though I realize Python is abbreviated `py`)

Reply:
The same goes below for paths.

Reply:
When change to java I see ouput: "make: `java' is up to date.", changed to javas

Reply:
Ah, really? Hmm. (Oh, it might be getting confused if java is both a directory and a target…in that case, I think `j` is OK as a target? So long as paths use `java`.)

Reply:
You need too mark the target as `PHONY` to avoid having Make confuse it with a file or directory.
See https://makefiletutorial.com/#phony

Reply:
Updated

> It might be worth writing a custom Sphinx extension like we did for C++ for these examples, especially since we're apparently duplicating code between examples and actual Java code.
> 
> For instance, maybe we could have something that accepts a file containing Java code, and the block is a series of JShell statements. Then for a docs build, it would just render the Java code and the JShell session. For a test build, it would run the Java code, then run the JShell statements, and check that the output matches what's expected.

In this case >jshell offer a "/edit" command when we could paste all our util code to define methods that then could be used for cookbook recipes or run all the cookbook code and then only validate output for /edit code and also /reset to do it again